### DISCUSSION TITLE
Issue to execute script tag after page transition

### DISCUSSION THREAD

**Discussion by mathieupreaud** [swup/swup#632] [Help] [answered]
I'm building a Wordpress website and I'm using Swup for page transition.

In one of the page I also have an Ajax call to load more posts with a button. To connect the loop with Ajax I declare some variables inside a script tag on the same page. Here it is:

```
<script class="scripts">
   var current_post_type = '<?php echo $current_post_type ?>',
   current_term_id = '<?php echo $current_term_id ?>',
   posts_myajax = '<?php echo serialize( $loop->query_vars ) ?>',
   current_page_myajax = 1,
   max_page_myajax = <?php echo $loop->max_num_pages ?>,
   ajax_url = '<?php echo esc_url( home_url( '/' ) ); ?>wp-admin/admin-ajax.php'
</script>

```
The Swup JS and the Ajax function to load more posts:

```
function init() {

	if (document.querySelector('.js-btn-load-more')) {

		const loaderBtn = document.querySelectorAll('.js-btn-load-more');
		var i;
		for( i = 0; i < loaderBtn.length; i++ ){
			loaderBtn[i].addEventListener('click', function(){

				const containerResponse = document.querySelector(".js-ajax-response");

				const data = new FormData();

				data.append( 'action', 'load_archive' );
				data.append( 'archive_post_type', current_post_type );
				data.append( 'taxonomy_term_id', current_term_id );
				data.append( 'query', posts_myajax );
				data.append( 'page', current_page_myajax );

				fetch(ajax_url, {
				method: "POST",
				body: data
				})
				.then((response) => response.text())
				.then((data) => {
				if (data) {
					containerResponse.innerHTML += data;
					
					current_page_myajax++;
				}
				})
				.catch((error) => {
				});
			});
		};
	}
	
}

init();

const options = {
	animationSelector: '[class*="o-page-transition-"]',
	linkSelector: 'a[href^="' +  window.location.origin + '"]:not([data-no-swup]), a[href^="/"]:not([data-no-swup]), a[href^="#"]:not([data-no-swup])',
	containers: [".js-content"]
};
const swup = new Swup(options);

swup.on('contentReplaced', init);
```

The issue is, when I navigate to the page via the Ajax page transition and then I click the button to load more posts, the console displays an error telling it can't find the variables from the script tag. If I load the page directly the Ajax load more function works fine.

Another thing, if I include this jQuery code to reload the script tag inside the init() function, it works. However I don't want to use jQuery just for this. I would prefer to solve it using plain javascript.

`$('.scripts').remove().appendTo('.container-scripts');`

I've been reading the Swup documentation about reloading scripts but I can't figure it out how to resolve the issue.

Thank you.

**Comment by daun** [comment, marked as answer]
The [scripts plugin](https://github.com/swup/scripts-plugin) should do exactly that for you, re-run inline script tags.

Alternatively, you could store your data on data attributes of an element that gets replaced by swup normally — and in your init script, read the data from that element:

```php
<div data-current-post-type="<?php echo $current_post_type ?">
```

```js
const current_post_type = document.querySelector('[data-current-post-type]').dataset.currentPostType;

**Reply by mathieupreaud** [reply]
Thank you @daun I like the idea to store data on data attributes. However, it creates an issue in the Ajax loop when I click the load more button. It always returns the same posts, that's weird.

Here is the JS code I added in my init() function:

```
const el = document.querySelector("#loader");

const current_post_type = el.dataset.currentPostType;
const current_term_id = el.dataset.currentTermId;
const posts_myajax = el.dataset.postsMyajax;
const current_page_myajax = el.dataset.currentPageMyajax;
const max_page_myajax = el.dataset.maxPageMyajax;
const ajax_url = el.dataset.ajaxUrl;
```

And the code inside the PHP template:

```
<div 
    id="loader" 
    data-current-post-type="<?php echo $current_post_type ?>" 
    data-current-term-id="<?php echo $current_term_id ?>" 
    data-posts-myajax="<?php echo serialize( $loop->query_vars ) ?>"
    data-current-page-myajax="1"
    data-max-page-myajax="<?php echo $loop->max_num_pages ?>"
    data-ajax-url="<?php echo esc_url( home_url( '/' ) ); ?>wp-admin/admin-ajax.php"
></div>
```

I don't know if I did something wrong.

EDIT

I resolved the issue adding let in state of const for:

`let current_page_myajax = el.dataset.currentPageMyajax;`
